dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -43,6 +43,11 @@ def __enter__(self):
                 TextIOWrapper(self._filepath_or_buffer, encoding=self._encoding)
             self._is_wrapped = True
 
+        elif isinstance(self._filepath_or_buffer, StringIO) \
```

Comment:
in what case does this occur? This seems a bit backwards to then convert it to bytes. 

Maybe the code elsewhere is wrong, but `rb` + `StringIO` shouldn't occur in our code.